app.py

```diff
@@ -328,12 +345,145 @@ def parse():
         'server_error': response.get('Error'),
         'version': version(),
         'username': username,
-        'is_test': 1 if os.getenv ('IS_TEST_ENV') else None,
+        'feedback_level': session['error_level'] if lang in supported_lang else None,
+        'GFM': True if lang in supported_lang else False,
+        'is_test': 1 if os.getenv('IS_TEST_ENV') else None,
         'adventure_name': body.get('adventure_name', None)
     })
 
     return jsonify(response)
 
+
+def gradual_feedback_model(code, level, gradual_feedback, language, E, hedy_exception):
+    response = {}
+    response['prev_feedback_level'] = session['error_level']
+    response['prev_similar_code'] = session['similar_code']
+
+    if session['code'] == code:
+        response["Feedback"] = gradual_feedback["Identical_code"]  # Don't raise the feedback level!
+        response["Duplicate"] = True
+    else:
+        response["Duplicate"] = False
+        if session['error_level'] < 5:  # Raise feedback level if is it not 5 (yet)
+            session['error_level'] = session['error_level'] + 1
```

Comment:
Same here and on every other session setting, so that it will work with A/B testing.

Reply:
Fixed!